src/dodal/devices/oav/pin_image_recognition/__init__.py

```diff
@@ -50,38 +51,30 @@ def __init__(self, prefix: str, name: str = ""):
         self._prefix: str = prefix
         self._name = name
 
-        self.triggered_tip = create_soft_signal_r(Tip, "triggered_tip", self.name)
-        self.triggered_top_edge = create_soft_signal_r(
+        self.triggered_tip, _ = soft_signal_r_and_setter(
+            Tip, "triggered_tip", self.name
+        )
+        self.triggered_top_edge, _ = soft_signal_r_and_setter(
             NDArray[np.uint32], "triggered_top_edge", self.name
         )
-        self.triggered_bottom_edge = create_soft_signal_r(
+        self.triggered_bottom_edge, _ = soft_signal_r_and_setter(
```

Comment:
As discussed, we can make a separate issue to make sure these signals use their setters rather than the backends in order to set values. Also, there may be scope to have a regular `soft_signal_r` function in ophyd-`async`